src/query/storages/hive/src/hive_table_source.rs

```diff
@@ -140,9 +161,28 @@ impl Processor for HiveTableSource {
 
     async fn async_process(&mut self) -> Result<()> {
         match std::mem::replace(&mut self.state, State::Finish) {
-            State::ReadData(part) => {
-                let (meta, chunks) = self.block_reader.read_columns_data(part.clone()).await?;
-                self.state = State::Deserialize(meta, chunks, part);
+            State::ReadMeta(part) => {
+                let part = HivePartInfo::from_part(&part)?;
+                let file_meta = self.block_reader.read_meta_data(&part.filename).await?;
```

Comment:
Can we cache this? If the file is splitted into multi partitions, we just need to read once for this file.

Reply:
we have a BloomIndexMetaCache, which caches `FileMetaData`, but is dedicated to bloom filter index files now.

https://github.com/datafuselabs/databend/blob/1b576fbe78552ca5d024205ee356af70b5b9edf1/src/query/storages/fuse-meta/src/caches/cache.rs#L41.

https://github.com/datafuselabs/databend/blob/26c74149b6b97b7152288b41ea0564114d617aaf/src/query/storages/fuse-meta/src/caches/memory_cache.rs#L50

maybe we could do the refactor in another dedicated PR?

Reply:
ok

Reply:
> Can we cache this? If the file is splitted into multi partitions, we just need to read once for this file.

yes, we need to do more works,  like try allocate partitions belong to one file to one worker, let's do it in another PR?